app/controllers/api/mixins/genealogy.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module Mixins
+    module Genealogy
+      RELATIONSHIP_COLLECTIONS = %w[vms templates].freeze
+      VALID_EDIT_ATTRS = %w[description name child_resources parent_resource].freeze
+
+      def tag_specified(id, data)
```

Comment:
I think all of these methods should be private so they are not exposed accidentally to the controller/router.